examples/solvers/adupdates_tomography.py

```diff
@@ -0,0 +1,151 @@
+regularization problem we are solving is therefore
+
+..math::
+    \min_{x\in \mathbb{R}^n_+} D(x) + \sum_{i = 1}^d (\| \partial_{i, 1} x \|_1
+    + \| \partial_{i, 2} x \|_1.
+
+Here, :math:`\partial_{i, 1}` and :math:`\partial_{i, 2}` contain the even and
+odd components, respectively, of the discretized :math:`i`th partial
+derivative, and :math:`d` is the dimension of the tomography problem. In this
+example, :math:`d = 2`. We solve the problem with the alternating dual updates
+method. For further details, see
+`[MF2015] <http://ieeexplore.ieee.org/document/7271047/>`_.
+
+References
+----------
+[MF2015] McGaffin, M G, and Fessler, J A. *Alternating dual updates
+algorithm for X-ray CT reconstruction on the GPU*. IEEE Transactions
+on Computational Imaging, 1.3 (2015), pp 186--199.
+"""
+
+import numpy as np
+import odl
+
+# Some constants we shall use in the sequel to avoid magic numbers.
+DIMENSIONS = 40.0     # Dimensions of reconstruction.
+RESOLUTION = 1024     # Discretization level of reconstruction space.
+SPLIT_METHOD = 'interlaced'  # How to split the data?
+SPLIT_NUMBER = 20     # How many pieces of data?
+NOISE_LEVEL = 5.0     # How much noise to add to the sinograms
+DATA_FIT_WEIGHT = 1.0  # Regularization parameter (coefficient of the data fit)
+EPSILON = 1e-6        # make sure that we do not divide by zero by adding this
+STEPSIZE = 1.         # Stepsize of the outer iterations
+NITER = 5             # Number of iterations
+RANDOM = True         # Choose the oder of the inner iterations at random?
+
+# --- Create simulated data (phantom) ---
+
+# Reconstruction space: Set of two-dimensional quadratic images with
+# given resolution.
+reconstruction_space = odl.uniform_discr(min_pt=[-DIMENSIONS, -DIMENSIONS],
+                                         max_pt=[DIMENSIONS, DIMENSIONS],
+                                         shape=[RESOLUTION, RESOLUTION])
+phantom = odl.phantom.shepp_logan(reconstruction_space, modified=True)
+
+# Create the forward operators. They correspond to a fully sampled parallel
+# beam geometry.
+geometry = odl.tomo.parallel_beam_geometry(reconstruction_space)
+
+if SPLIT_METHOD == 'block':
+    # Split the data into blocks:
+    # 111 222 333
+    ns = geometry.angles.size // SPLIT_NUMBER
+    ray_trafos = [odl.tomo.RayTransform(reconstruction_space,
+                                        geometry[i * ns:(i + 1) * ns])
+                  for i in range(SPLIT_NUMBER)]
+
+elif SPLIT_METHOD == 'interlaced':
+    # Split the data into slices:
+    # 123 123 123
+    ray_trafos = [odl.tomo.RayTransform(reconstruction_space,
+                                        geometry[i::SPLIT_NUMBER])
+                  for i in range(SPLIT_NUMBER)]
+else:
+    raise ValueError('unknown data split`{}`: typo?'.format(SPLIT_METHOD))
+
+# Create the artificial data.
+data_spaces = [op.range for op in ray_trafos]
+data = [op(phantom) + NOISE_LEVEL * odl.phantom.white_noise(ds)
```

Comment:
Here, for instance, there is no advantage of having `NOISE_LEVEL` as a constant above. The value is still entirely magical, and to change it, people will have to navigate to the top of the file.